src/Carousel/index.scss

```diff
@@ -58,7 +62,11 @@
   .active.carousel-item-right {
     z-index: 0;
     opacity: 0;
-    transition: opacity 0s var(--pgn-transition-carousel-base);
+    transition:
+      opacity 0s var(--pgn-transition-carousel-base-property)
+      var(--pgn-transition-carousel-base-duration)
```

Comment:
0s in this context is the delay time before the transition starts for the opacity property. I think we shouldn't use the `var(--pgn-transition-carousel-base-delay)` variable here, nor should we use `var(--pgn-transition-carousel-base-duration)`, since this variable refers to the delay time for the `transition` property. I think if we want to get rid of the hardcoded `0s` value, we'll need to create a new token for that. Or we can leave the value as it is now, since there was no variable for it before.

Reply:
I'm not sure I'm fully following how this fits into the css `transition` syntax. On ([MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/transition)) the closest I can find is

```
/* property name | duration | easing function | delay */
transition: margin-right 4s ease-in-out 1s;
```

which before this change would be

|  |  |
| - | - |
| property name | `opacity` |
| duration | `0s` |
| easing function | part of `var(--pgn-transition-carousel-base)` |
| delay | part of `var(--pgn-transition-carousel-base)` |

but after this change we have

|  |  |
| - | - |
| property name | `opacity` |
| duration | `0s` |
| easing function | `var(--pgn-transition-carousel-base-property)` |
| delay | `var(--pgn-transition-carousel-base-duration)` |
| ? | `var(--pgn-transition-carousel-base-timing-function)` |
| ? | `var(--pgn-transition-carousel-base-delay)` |

is there something I'm missing here?

Reply:
Good catch! There is a missing comma here because we have two different transitions: a transition for the `opacity` property and a transition for `transform`.

Added `transition: opacity 0s, var(--pgn-transition-carousel-base);`

Reply:
With the token $value as an array, it'd end up looking something like this:

```scss
  :root {
    --pgn-transition-carousel-base):
      var(--pgn-transition-carousel-base-1-property) --pgn-transition-carousel-base-1-duration,
      var(--pgn-transition-carousel-base-2-property) --pgn-transition-carousel-base-2-duration --pgn-transition-carousel-base-2-timing-function --pgn-transition-carousel-base-2-delay;
  }
    
  transition: var(--pgn-transition-carousel-base);
```

Reply:
Added in [`24fd229` (#3222)](https://github.com/openedx/paragon/pull/3222/commits/24fd2297063f38641f376bbeadf4e8b98d16bfd8)

Also added automatic generation of the `abstraction-variables.css` file and added the correct order of imports in the `index.css` files for core and themes tokens.